src/handlers/default.ts

```diff
@@ -171,7 +175,24 @@ export class InngestCommHandler {
 
   public createHandler(): any {
     return async (req: Request, res: Response) => {
-      const reqUrl = new URL(req.originalUrl, req.hostname);
+      const hostname = req.hostname || req.headers["host"];
+      if (!hostname) {
+        console.error("Unable to determine your site URL to host Inngest.");
+        return res.status(500).json("Unable to determine your site URL to host Inngest.");
+      }
+
+      let scheme = "";
+      if (hostname.indexOf("://") === -1) {
+        scheme = (process.env.NODE_ENV === "development" || !req.connection.encrypted) ? "http://" : "https://";
```

Comment:
I couldn't find `req.connection` cited in Express and it's marked as [deprecated in the Node docs](https://nodejs.org/dist/latest-v16.x/docs/api/http.html#requestconnection) - Can we use `req.protocol` instead?

* https://expressjs.com/en/api.html#req.protocol
* https://nodejs.org/dist/latest-v16.x/docs/api/http.html#requestprotocol

Reply:
Done, thanks!